apis/keda/v1alpha1/scaledobject_webhook.go

```diff
@@ -183,6 +183,7 @@ func verifyReplicaCount(incomingSo *ScaledObject, action string, _ bool) error {
 	if err != nil {
 		scaledobjectlog.WithValues("name", incomingSo.Name).Error(err, "validation error")
 		metricscollector.RecordScaledObjectValidatingErrors(incomingSo.Namespace, action, "incorrect-replicas")
+		return err
```

Comment:
is it possible this may block previously valid SOs with min undefined and max set to 1?

if I'm reading the code correctly, this calls `CheckReplicaCountBoundsAreValid()` which in case of min being not defined sets it to 1 as min for HPA. The check would then be `1 < 1 == false` so it won't allow it.
https://github.com/kedacore/keda/blob/5c52d032931b8ecf855d0c298f8d5e48937aecd7/apis/keda/v1alpha1/scaledobject_types.go#L259-L268
https://github.com/kedacore/keda/blob/5c52d032931b8ecf855d0c298f8d5e48937aecd7/apis/keda/v1alpha1/scaledobject_types.go#L241-L247
https://github.com/kedacore/keda/blob/5c52d032931b8ecf855d0c298f8d5e48937aecd7/apis/keda/v1alpha1/scaledobject_types.go#L81

What am I missing? There surely is a test for this and given all tests are passing... I'm not realizing something trivially obvious :)

Reply:
ah, never mind, it is `min > max then error`, so it checks `1 > 1` which is `false` resulting in no error returned.